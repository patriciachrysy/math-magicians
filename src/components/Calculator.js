
class Calculator extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            screenValue: '0'
        }
    }

    render = () => {
        return <div class="calculator-frame"></div>
    }
}