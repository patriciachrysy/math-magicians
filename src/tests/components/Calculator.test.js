import React from 'react';
import renderer from 'react-test-renderer';

import { render, screen } from '@testing-library/react';
import Calculator from '../../components/Calculator';

it('renders correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render Calculator component', () => {
  render(<Calculator />);
  screen.debug();
});
