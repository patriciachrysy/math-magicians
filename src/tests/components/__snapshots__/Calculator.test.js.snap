// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="wrapper"
>
  <h2>
    Let‘s do some math!
  </h2>
  <div
    className="calculator-frame"
  >
    <div
      className="screen-frame"
    >
      <span>
        0
      </span>
    </div>
    <div
      className="keyboard-frame"
    >
      <button
        onClick={[Function]}
        type="button"
        value="AC"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="+/-"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="%"
      >
        %
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="÷"
      >
        ÷
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="7"
      >
        7
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="8"
      >
        8
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="9"
      >
        9
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="x"
      >
        x
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="4"
      >
        4
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="5"
      >
        5
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="6"
      >
        6
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="-"
      >
        -
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="1"
      >
        1
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="2"
      >
        2
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="3"
      >
        3
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="+"
      >
        +
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="0"
      >
        0
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="."
      >
        .
      </button>
      <button
        onClick={[Function]}
        type="button"
        value="="
      >
        =
      </button>
    </div>
  </div>
</div>
`;
